src/mongocxx/options/find_one_and_update.hpp

```diff
@@ -27,6 +26,17 @@ MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace options {
 
 ///
+/// Argument for find_one_and_update and find_one_and_replace to determine
+/// whether to return the document before or after modification.
+///
+/// @see https://docs.mongodb.org/manual/reference/method/db.collection.findAndModify/#return-data
+///
+enum class return_document {
```

Comment:
No, if we leave it then we're still having one class rely on the other's header.  I opt for moving this to a new header for common find_one_* options.  It will admittedly be a very short header file but it seems like a cleaner solution.